### DISCUSSION TITLE
Get the current status of a dynamic DNS entry?

### DISCUSSION THREAD

**Discussion by fdcastel** [troglobit/inadyn#455] [Q&A]
**TL;DR:** Is there a way to ask for `inadyn` the current status (`ok`, `needs-update`, `last-attempt-failed`, etc) of a configured dynamic DNS entry?

-----
Long version:

Vyatta-based routers have a useful command

```
show dns dynamic status
```

Which shows the current status of all your configured DDNS records:

```
interface    : eth0
ip address   : XXX.XXX.215.113
host-name    : MY_DOMAIN.COM
last update  : Mon Sep 25 10:10:51 2023
update-status: good
```

The important property here is `update-status`. Any eventual errors occurred in last update will show here.

Vyatta-based routers use `ddclient` for dynamic dns updates, which in its `cache` file for the interface saves the following information:

```
## ddclient-3.8.3
## last updated at Mon Sep 25 10:11:56 2023 (1695647516)
atime=0,custom=0,host=MY_DOMAIN.COM,ip=XXX.XXX.215.113,mtime=1695647451,static=0,status=good,warned-min-error-interval=0,warned-min-interval=0,wtime=30 MY_DOMAIN.COM
```

As we can see, the `show dns dynamic status` is a simply wrapper to show us the `cache` content from `ddclient`.

**Is there anything similar in `inadyn`?**

To the best of my knowledge, the only information it stores is the last ip (on `cache` file).

**Comment by fdcastel** [comment]
My 0.02: Any **stored** status information can be outdated in the next second.

But a kind of `--dry-run` argument can always be useful in a myriad of situations.

**Reply by troglobit** [reply]
There is nothing today, but I agree that it would be highly useful.  Inadyn caches the last sent IP address per provider+instance so we could use that + last modified timestamp on that file to generate something similar from `inadyn --show-status`, or similar.

**Reply by fdcastel** [reply]
👍 

Working on a proposal to extend this.

**Reply by troglobit** [reply]
Fantastic, thank you!  If you keep this up you'll make co-maintainer in no time ;-)

**Comment by fdcastel** [comment]
To be continued in #457 .
